test/compiler/inference.jl

```diff
@@ -4669,3 +4669,10 @@ bar47688() = foo47688()
 @test it_count47688 == 7
 @test isa(foo47688(), NTuple{6, Int})
 @test it_count47688 == 14
+
+# refine instantiation of partially-known NamedTuple that is known to be empty
+@test Base.return_types((Any,)) do Tpl
+    T = NamedTuple{(),Tpl}
+    nt = T(())
+    values(nt)
+end === Tuple{}
```

Comment:
Needs an `only`.

Also, IIUC correctly, this should equally handle the `NamedTuple{Tpl, Tuple{}}` case, so also add a test for that?